### PR TITLE
2.03 add iso date for baseline element

### LINKED ISSUE #364: Add new @iso-date attribute for the baseline element
Add a new attribute at `iati-activities/iati-activity/result/indicator/baseline/@iso-date`:

The attribute will have type `xsd:date` with occurrence rule `0..1`.

**Tasks:**

- [x] Add tests
- [x] Add `iati-activities/iati-activity/result/indicator/baseline/@iso-date`

### PR DESCRIPTION
**Pull request by allthatilk**
Fixes #364

### CODE REVIEW

File: tests/should-fail/iati-activities/iati-activity/result/indicator/baseline/dimension/err-not-iati-xml-incorrect-datatype.xml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0"?>
+
+<iati-activities version="xx">
+  <iati-activity> <!-- at least 1 -->
+    <iati-identifier></iati-identifier>  <!-- 1 and only 1-->
```
Comment (dalepotter) [inline tests/should-fail/iati-activities/iati-activity/result/indicator/baseline/dimension/err-not-iati-xml-incorrect-datatype.xml]: All these comments to be removed in this and other test case files :-)

File: tests/should-fail/iati-activities/iati-activity/result/indicator/baseline/dimension/err-not-iati-xml-incorrect-datatype.xml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    </activity-date>
+    <result type="xx">
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <indicator measure="xx">
+        <title>
+          <narrative>Sample text</narrative>
+        </title>
+        <baseline year="2017" value= "xx" iso-date="123456"><!-- iso-date attribute included but invalid date given -->
```
Comment (dalepotter) [inline tests/should-fail/iati-activities/iati-activity/result/indicator/baseline/dimension/err-not-iati-xml-incorrect-datatype.xml]: Prepend comment with `FAIL:`.

File: tests/should-pass/iati-activities/iati-activity/result/indicator/baseline/attribute-exists-for-element.xml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    </activity-date>
+    <result type="xx">
+      <title>
+        <narrative>Sample text</narrative>
+      </title>
+      <indicator measure="xx">
+        <title>
+          <narrative>Sample text</narrative>
+        </title>
+        <baseline year="2017" value= "xx" iso-date="2017-11-16"><!-- iso-date attribute included -->
```
Comment (dalepotter) [inline tests/should-pass/iati-activities/iati-activity/result/indicator/baseline/attribute-exists-for-element.xml]: Suggest change comment to `PASS: iso-date attribute included using correct datetime format.`

**Review by dalepotter — CHANGES_REQUESTED** [review]
A couple small things.

Also, is there a test case file somewhere for the element with no `iso-date` attribute?

**Comment by allthatilk** [comment]
I assumed that no iso-date would be covered by past test data for baseline.

**Review by dalepotter — APPROVED** [review]
thumbs up

**Pull request merged**